server/service/appconfig.go

```diff
@@ -39,6 +43,23 @@ type appConfigResponse struct {
 	Err error `json:"error,omitempty"`
```

Comment:
Could `Err` still be a field on `appConfigResponse`? I feel like this is where it should belong, but there may be technical reasons not to put it there with the custom marshaling. Nothing critical if that's not possible.

Reply:
totally possible, but AFAIK, we will have to do another unmarshalling pass:

```go
	if err := json.Unmarshal(data, &r.Err); err != nil {
		return err
	}
```

what do you think?

Reply:
Yeah, I was gonna say outside of tests we only ever marshal that struct (since it's the response payload), but looks like it is used in `service/client_appconfig.go` for fleetctl. Might not be worth the trouble.